pkg/minikube/machine/delete.go

```diff
@@ -75,18 +75,26 @@ func DeleteHost(api libmachine.API, machineName string) error {
 		return mcnerror.ErrHostDoesNotExist{Name: machineName}
 	}
 
-	// This is slow if SSH is not responding, but HyperV hangs otherwise, See issue #2914
+	// Hyper-V requires special care to avoid ACPI and file locking issues
 	if host.Driver.DriverName() == driver.HyperV {
-		if err := trySSHPowerOff(host); err != nil {
-			glog.Infof("Unable to power off minikube because the host was not found.")
+		if err := StopHost(api, machineName); err != nil {
+			glog.Warningf("stop host: %v", err)
 		}
-		out.T(out.DeletingHost, "Successfully powered off Hyper-V. minikube driver -- {{.driver}}", out.V{"driver": host.Driver.DriverName()})
+		// Hack: give the Hyper-V VM more time to stop before deletion
+		time.Sleep(1 * time.Second)
 	}
 
 	out.T(out.DeletingHost, `Deleting "{{.profile_name}}" in {{.driver_name}} ...`, out.V{"profile_name": machineName, "driver_name": host.DriverName})
 	if err := host.Driver.Remove(); err != nil {
-		return errors.Wrap(err, "host remove")
+		glog.Warningf("remove failed, will retry: %v", err)
```

Comment:
nit: how about using retry.Expo ( and set the max number of retry to 2 or 3)

Reply:
I considered it, but feel like it may provide an awkward experience to retry more than once. Will reconsider based on user reports.